Gems/Atom/Feature/Common/Assets/Shaders/Reflections/ReflectionScreenSpaceRayTracingCommon.azsli

```diff
@@ -0,0 +1,698 @@
+    if (!CheckLightVisibility(lightingData, dirToLight, tmax))
+    {
+        return;
+    }
+
+    // smoothly adjusts the light intensity so it reaches 0 at light.m_attenuationRadius distance
+    float radiusAttenuation = 1.0f - (falloff * falloff);
+    radiusAttenuation = radiusAttenuation * radiusAttenuation;
+        
+    // find the distance to the closest point on the disk
+    float distanceToPlane = dot(posToLight, -light.m_direction);
+    float distanceToPlane2 = distanceToPlane * distanceToPlane;
+    float pointOnPlaneToLightDistance = sqrt(d2 - distanceToPlane2); // pythagorean theorem 
+    float pointOnPlaneToDiskDistance = max(pointOnPlaneToLightDistance - light.m_diskRadius, 0.0f); 
+    float distanceToDisk2 = pointOnPlaneToDiskDistance * pointOnPlaneToDiskDistance + distanceToPlane2;
+
+    // update the light direction based on the edges of the disk as visible from this point instead of the center.
+    float3 pointOnPlane = -light.m_direction * distanceToPlane;
+    float3 pointOnPlaneToLightDir = normalize(posToLight - pointOnPlane);
+    float3 nearSideDir = normalize(pointOnPlane + pointOnPlaneToLightDir * (pointOnPlaneToLightDistance - light.m_diskRadius));
+    float3 farSideDir = normalize(pointOnPlane + pointOnPlaneToLightDir * (pointOnPlaneToLightDistance + light.m_diskRadius));
+    dirToLight = normalize((nearSideDir + farSideDir) * 0.5f);
+
+    // standard quadratic falloff
+    distanceToDisk2 = max(0.001f * 0.001f, distanceToDisk2); // clamp the light to at least 1mm away to avoid extreme values.
+    float3 lightIntensity = (light.m_rgbIntensityCandelas / distanceToDisk2) * radiusAttenuation * angleFalloff;
+
+    // adjust brightness based on the disk size relative to its distance,
+    // the larger the disk is relative to the surface point, the dimmer it becomes.
+    // 0 radius disks are unaffected.
+    lightIntensity /= ((light.m_diskRadius / distanceToPlane) + 1.0f);
+
+    diffuse += DiffuseLambertian(lightingData.m_albedo, lightingData.m_normal, dirToLight, lightingData.m_diffuseResponse) * lightIntensity;
+
+    if (applySpecular)
+    {
+        specular += SpecularGGX(lightingData.m_dirToCamera, dirToLight, lightingData.m_normal, lightingData.m_specularF0, lightingData.m_NdotV, lightingData.m_roughnessA2, 1.0f) * lightIntensity;
+    }
+}
+
+//! Adjust the intensity of specular light based on the radius of the light source and roughness of the surface to approximate energy conservation.
+float GetIntensityAdjustedByRadiusAndRoughness(float roughnessA, float radius, float distance2)
+{
+    float roughnessAdjusted = saturate(roughnessA + (radius / (3.0 * sqrt(distance2))));
+    float intensityNormalization = max(roughnessA, 0.001f) / roughnessAdjusted;
+    return intensityNormalization * intensityNormalization;
+}
+
+void ApplyCapsuleLight(RayTracingSceneSrg::CapsuleLight light, RayTracingLightingData lightingData, bool applySpecular, out float3 diffuse, out float3 specular)
```

Comment:
I see a lot of duplicated code here vs ApplyCapsuleLight. Probably true for other functions here